src/Checkbox/Input.js

```diff
@@ -1,27 +1,60 @@
-import React, { Fragment } from 'react'
+import React, { Component, Fragment, createRef } from 'react'
 import propTypes from 'prop-types'
 import css from 'styled-jsx/css'
 
-export const Input = ({ onChange, checked, disabled, name }) => (
-    <Fragment>
-        <input
-            type="checkbox"
-            name={name}
-            onChange={onChange}
-            checked={checked}
-            disabled={disabled}
-        />
+export class Input extends Component {
+    ref = createRef()
 
-        <style jsx>{`
-            input {
-                display: none;
-            }
-        `}</style>
-    </Fragment>
-)
+    componentDidMount() {
+        if (this.props.focus) {
+            this.ref.current.focus()
+        }
+    }
+
+    render() {
+        const {
+            onChange,
+            checked,
+            disabled,
+            name,
+            value,
+            onFocus,
+            onBlur,
+        } = this.props
+
+        return (
+            <Fragment>
+                <input
+                    tabIndex="0"
+                    ref={this.ref}
+                    type="checkbox"
+                    name={name}
+                    value={value}
+                    onChange={onChange}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                    checked={checked}
+                    disabled={disabled}
+                />
+
+                <style jsx>{`
+                    input {
+                        display: block;
+                        opacity: 0;
+                        width: 0;
+                        height: 0;
+                        outline: 0;
+                    }
+                `}</style>
+            </Fragment>
+        )
+    }
+}
 
 Input.propTypes = {
     onChange: propTypes.func.isRequired,
+    onFocus: propTypes.func.isRequired,
+    onBlur: propTypes.func.isRequired,
```

Comment:
Why are these required on the Input but not on the Checkbox?

Reply:
The `Input` will have to update the parent component (the `Checkbox` in this case) about the focus state, but the app that implements the checkbox might not care about the focus state at all.
If added the `isRequired` to the `SelectField`'s `Select` component to make it consistent across our input components.

Reply:
Ok.